operate/schema/src/main/java/io/camunda/operate/schema/IndexMappingDifference.java

```diff
@@ -0,0 +1,215 @@
+    this.entriesDiffering = entriesDiffering;
+    return this;
+  }
+
+  public static IndexMappingDifference from(MapDifference<String, Object> difference) {
+    return new IndexMappingDifference()
+        .setEqual(difference.areEqual())
+        .setEntriesOnlyOnLeft(
+            difference.entriesOnlyOnLeft().entrySet().stream()
+                .map(getIndexMappingProperty())
+                .collect(Collectors.toSet()))
+        .setEntriesOnlyOnRight(
+            difference.entriesOnlyOnRight().entrySet().stream()
+                .map(getIndexMappingProperty())
+                .collect(Collectors.toSet()))
+        .setEntriesInCommon(
+            difference.entriesInCommon().entrySet().stream()
+                .map(getIndexMappingProperty())
+                .collect(Collectors.toSet()))
+        .setEntriesDiffering(
+            difference.entriesDiffering().entrySet().stream()
+                .map(
+                    entry ->
+                        new PropertyDifference()
+                            .setName(entry.getKey())
+                            .setLeftValue(
+                                new IndexMappingProperty()
+                                    .setName(entry.getKey())
+                                    .setTypeDefinition(entry.getValue().leftValue()))
+                            .setRightValue(
+                                new IndexMappingProperty()
+                                    .setName(entry.getKey())
+                                    .setTypeDefinition(entry.getValue().rightValue())))
+                .collect(Collectors.toSet()));
+  }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    final IndexMappingDifference that = (IndexMappingDifference) o;
+    return equal == that.equal
+        && Objects.equals(entriesOnlyOnLeft, that.entriesOnlyOnLeft)
+        && Objects.equals(entriesOnlyOnRight, that.entriesOnlyOnRight)
+        && Objects.equals(entriesInCommon, that.entriesInCommon)
+        && Objects.equals(entriesDiffering, that.entriesDiffering);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(
+        equal, entriesOnlyOnLeft, entriesOnlyOnRight, entriesInCommon, entriesDiffering);
+  }
+
+  @Override
+  public String toString() {
+    return "IndexMappingDifference{"
+        + "equal="
+        + equal
+        + ", entriesOnlyOnLeft="
+        + entriesOnlyOnLeft
+        + ", entriesOnlyOnRight="
+        + entriesOnlyOnRight
+        + ", entriesInCommon="
+        + entriesInCommon
+        + ", entriesDiffering="
+        + entriesDiffering
+        + '}';
+  }
+
+  public static Function<Entry<String, Object>, IndexMappingProperty> getIndexMappingProperty() {
+    return entry ->
+        new IndexMappingProperty().setName(entry.getKey()).setTypeDefinition(entry.getValue());
+  }
```

Comment:
Is it necessary that this returns a function or could this method just implement the plain logic and you use a function reference to it?